indexer/sql/src/non_blocking_indexer.rs

```diff
@@ -192,15 +198,122 @@ impl NonBlockingIndexer {
 
         Ok(block_to_index.1)
     }
+
+    /// Wait for all blocks to be indexed
+    pub fn wait_for_finish(&self) {
+        for _ in 0..10 {
+            if self.blocks.lock().expect("Can't lock blocks").is_empty() {
+                break;
+            }
+
+            sleep(Duration::from_millis(10));
+        }
+    }
+}
+
+// ------------------------------- DB Related ----------------------------------
+
+impl NonBlockingIndexer {
+    /// Delete a block and its related content from the database
+    pub fn delete_block_from_db(&self, block_height: u64) -> error::Result<()> {
+        self.handle.block_on(async move {
+            let db = self.context.db.begin().await?;
+            entity::blocks::Entity::delete_block_and_data(&db, block_height).await?;
+            db.commit().await?;
+
+            Ok::<(), sea_orm::error::DbErr>(())
+        })?;
+        Ok(())
+    }
+}
+
+impl NonBlockingIndexer {
+    /// Where will this block be temporarily saved on disk
+    pub fn block_filename(&self, block_height: u64) -> PathBuf {
+        let directory = self.indexer_path.block_path();
+        let filename = format!("{block_height}");
+        directory.join(filename)
+    }
+
+    /// Load all existing tmp files and ensure they've been indexed
+    fn ensure_previous_indexed_blocks(&self, latest_block_height: u64) -> error::Result<()> {
```

Comment:
Renamed to `index_previous_unindexed_blocks`

> Also, what happend if we remove from the db a block that is the latest indexed? That block is not reindexed again right?

No block should ever be removed from the DB until we implement an indexer prune which nodes could run to clean up space. It's only removed in the test to build the context of that test.

To reindex missing blocks, you need to save blocks on disk with `set_keep_blocks()` and I'll need to write a binary (coming later) to "reindex" all on-disk blocks. This will be needed once we're in production and feature changes in the indexer code requires to reindex previous blocks. It's rare, but it will happen and so why I have that `keep_blocks` feature.